modules/machineset-capacity-reservation.adoc

```diff
@@ -0,0 +1,161 @@
+ifdef::aws[{aws-short} offering, see link:https://docs.aws.amazon.com/en_us/AWSEC2/latest/UserGuide/capacity-reservation-overview.html[On-Demand Capacity Reservations and Capacity Blocks for ML] in the {aws-short} documentation.]
+
+ifdef::azure[]
+[NOTE]
+====
+You cannot change an existing Capacity Reservation configuration for a machine set.
+To use a different Capacity Reservation group, you must replace the machine set and the machines that the previous machine set deployed.
+====
+endif::azure[]
+
+.Prerequisites
+
+* You have access to the cluster with `cluster-admin` privileges.
+* You installed the {oc-first}.
+ifdef::azure[]
+* You created a Capacity Reservation group.
+For more information, see link:https://learn.microsoft.com/en-us/azure/virtual-machines/capacity-reservation-create[Create a Capacity Reservation] in the {azure-full} documentation.
+endif::azure[]
+ifdef::aws[]
+* You purchased an On-Demand Capacity Reservation or Capacity Block for ML.
+For more information, see link:https://docs.aws.amazon.com/en_us/AWSEC2/latest/UserGuide/capacity-reservation-overview.html[On-Demand Capacity Reservations and Capacity Blocks for ML] in the {aws-short} documentation.
+endif::aws[]
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following section under the `providerSpec` field:
++
+--
+.Sample configuration
+[source,yaml]
+----
+tag::compute[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+end::compute[]
+tag::controlplane[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+end::controlplane[]
+# ...
+spec:
+  template:
+tag::compute[]
+    spec:
+      providerSpec:
+        value:
+ifdef::azure[]
+          capacityReservationGroupID: <capacity_reservation_group> # <1>
+endif::azure[]
+ifdef::aws[]
+          capacityReservationId: <capacity_reservation> # <1>
+          marketType: <market_type> # <2>
+endif::aws[]
+end::compute[]
+tag::controlplane[]
+    machines_v1beta1_machine_openshift_io:
+      spec:
+        providerSpec:
+          value:
+ifdef::azure[]
+            capacityReservationGroupID: <capacity_reservation_group> # <1>
+endif::azure[]
+ifdef::aws[]
+            capacityReservationId: <capacity_reservation> # <1>
+            marketType: <market_type> # <2>
+endif::aws[]
+end::controlplane[]
+# ...
+----
+<1> Specify the ID of the 
+ifdef::azure[Capacity Reservation group]
+ifdef::aws[Capacity Block for ML or On-Demand Capacity Reservation]
+that you want the machine set to deploy machines on.
+ifdef::aws[]
+<2> Specify the market type to use.
+The following values are valid:
+`CapacityBlock`:: Use this market type with Capacity Blocks for ML.
+`OnDemand`:: Use this market type with On-Demand Capacity Reservations.
+tag::compute[]
+`Spot`:: Use this market type with Spot Instances.
```

Comment:
@huali9 I noticed in your testing that you are showing this configuration:
```
      marketType: Spot
      spotMarketOptions: {}
```
Currently, [the Spot Instance docs](https://88248--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-aws.html#machineset-creating-non-guaranteed-instance_creating-machineset-aws) only have `spotMarketOptions: {}`. Do they need the `marketType: Spot` parameter too?

Reply:
ah, never mind - I misread one note that `spotMarketOptions: {}` still works by itself. I guess i have a different question: is there a reason the user might want to specify `marketType: Spot` if it is purely optional?

Reply:
Yes, all the three options behave the same, they all create spot instance.
```
      marketType: Spot
      spotMarketOptions: {}
```
or
`      marketType: Spot`
or
`      spotMarketOptions: {}`